src/cocotb/task.py

```diff
@@ -217,16 +217,20 @@ def __repr__(self) -> str:
 
         if self._state is _TaskState.RUNNING:
             return f"<{self._name} running coro={coro_name}()>"
-        elif self._state is _TaskState.FINISHED:
-            return f"<{self._name} finished coro={coro_name}() outcome={self._outcome}>"
+        elif self._state in (_TaskState.FINISHED, _TaskState.ERRORED):
+            return (
+                f"<{self._name} finished coro={coro_name}() outcome={self._outcome!r}>"
+            )
```

Comment:
I can split this into "finished" and "errored" if desired. It would make it more obvious that it ended with an exception vs returned one.